src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,257 @@
+      }
+    }
+    return isParentExpanded;
+  }, [expandedNodes, parentNode]);
+
+  const [isChecked, setIsChecked] = React.useState(
+    React.isValidElement(nodeCheckbox) ? nodeCheckbox.props['checked'] : false,
+  );
+
+  const selectedNodes = React.useMemo(() => {
+    return context?.selectedNodes ?? [];
+  }, [context?.selectedNodes]);
+
+  const isSelected = React.useMemo(() => {
+    return selectedNodes.findIndex((id) => id === nodeId) != -1;
+  }, [selectedNodes, nodeId]);
+
+  const onNodeClick = () => {
+    if (isDisabled) {
+      return;
+    }
+    if (isSelected) {
+      context?.setSelectedNode?.(
+        selectedNodes.filter((item) => item != nodeId),
+      );
+    } else {
+      context?.setSelectedNode?.([nodeId]);
+    }
+    onNodeSelected?.(nodeId, !isSelected);
+  };
+
+  return (
+    <>
+      <li role='treeitem' id={nodeId} aria-expanded={isExpanded} {...rest}>
+        {visible && (
+          <div
+            className={cx(
+              'iui-tree-node',
+              {
+                'iui-active': isSelected,
+                'iui-disabled': isDisabled,
+              },
+              className,
+            )}
+            style={styleLevel}
+            onClick={() => onNodeClick()}
+            tabIndex={isSelected ? 0 : -1}
+          >
+            {nodeCheckbox && React.isValidElement(nodeCheckbox)
+              ? React.cloneElement(nodeCheckbox, {
+                  className: 'iui-tree-node-checkbox',
+                  disabled: isDisabled,
+                  checked: isChecked,
+                  onClick: () => {
+                    setIsChecked(!isChecked);
+                    onNodeCheckboxSelected?.(nodeId, !isChecked);
+                  },
+                })
+              : nodeCheckbox}
+            <div className='iui-tree-node-content'>
+              {subNodes && subNodes.length > 0 && (
+                <IconButton
+                  styleType='borderless'
+                  size='small'
+                  onClick={(e) => onExpandChanged(e)}
+                  disabled={isDisabled}
+                  tabIndex={-1}
+                >
+                  <SvgChevronRight
+                    className={cx('iui-tree-node-content-expander-icon', {
+                      'iui-tree-node-content-expander-icon-expanded': isExpanded,
+                    })}
+                  />
+                </IconButton>
+              )}
+              {icon &&
+                React.cloneElement(icon, {
+                  className: cx(
+                    'iui-tree-node-content-icon',
+                    icon.props.className,
+                  ),
+                })}
+              <span className='iui-tree-node-content-label'>
+                <div className='iui-tree-node-content-title'>{label}</div>
+                <div className='iui-tree-node-content-caption'>{sublabel}</div>
+              </span>
+            </div>
+            {children}
```

Comment:
Maybe we should move `children` inside `iui-tree-node-content` div.

Reply:
Moved